cmd/data-scanner.go

```diff
@@ -103,6 +103,33 @@ func (s *safeDuration) Get() time.Duration {
 	return s.t
 }
 
+type safeTime struct {
+	sync.Mutex
+	t time.Time
+}
+
+func (s *safeTime) Update(t time.Time) {
+	s.Lock()
+	defer s.Unlock()
+	s.t = t
+}
+
+func (s *safeTime) Get() time.Time {
+	s.Lock()
+	defer s.Unlock()
+	return s.t
+}
+
+func getCycleScanMode() madmin.HealScanMode {
+	bitrotCycle := globalHealConfig.BitrotScanCycle()
+	if bitrotCycle == 0 || (time.Since(globalBootTime) > bitrotCycle && time.Since(globalLastBitrotScan.Get()) > bitrotCycle) {
+		globalLastBitrotScan.Update(time.Now())
+		return madmin.HealDeepScan
+	}
+
+	return madmin.HealNormalScan
+}
```

Comment:
Nothing of this makes sense to me.

`globalBootTime`? So if you are a nice boy and update often, you never get it? Persist the value.

You are aware that it takes 512 complete cycles before all files have been heal checked?

Reply:
> globalBootTime? So if you are a nice boy and update often, you never get it? Persist the value.

Yes, you are right, will fix it

> You are aware that it takes 512 complete cycles before all files have been heal checked?

This function is actually called before each cycle. 512 is not related to this.

Reply:
> This function is actually called before each cycle. 512 is not related to this.

Oh I see what you mean.. Let me think about it